pycloudlib/ec2/instance.py

```diff
@@ -355,24 +374,34 @@ def remove_network_interface(self, ip_address):
         NIC.
         """
         # Get the NIC from the IP
-        nic = [
-            nic
-            for nic in self._instance.network_interfaces
-            if nic.private_ip_address == ip_address
-        ][0]
+        nic = next(
+            (
+                nic
+                for nic in self._instance.network_interfaces
+                if nic.private_ip_address == ip_address
+            ),
+            None,
+        )
+        if not nic:
+            self._log.debug(
+                "Not deleting NIC because no NIC with IP {} found."
+            )
+            return
```

Comment:
Is the reason we aren't realizing the full list of `nics_matching_ips` outside of the for loop below because we are expecting our self._instance.reload() to show the dwindling list of matching NICs over a period of loop iterations?

Reply:
We're not expecting a single IP to be attached to multiple NICs. I'm not sure if that's even possible, but the way the add_network_interface() code works, it should be 1:1. We loop through all of the nics (via list comprehension) because that's the only way to know if we have a nic that matches this particular IP, but we're only ever expecting 1 result from the loop.

Does that make sense? Do you think there's a way to make it clearer?

Reply:
Nah, it's just the same boilerplate code used a few lines below. Maybe separate that to a private function `_get_nic_for_ip` and call it here and within the for loop?